flatten-array.md

```diff
@@ -0,0 +1,19 @@
+The challenge is to write a function that accepts an arbitrarily-deep nested Array-like structure and returns a flattened structure with any null values removed without using the reduce method.
```

Comment:
Not all languages have a reduce method. We'll need a way to phrase this so that it can be used for any languages.

Likewise, I think maybe we should say _list_ rather than _Array-like_. In Go, for example, we would probably use slices, not arrays for this.